### PR TITLE
Your account page

### PR DESCRIPTION
**Pull request by dushan-madetech**
Your account page allows users to see and edit the data held about them

https://digital.dclg.gov.uk/jira/browse/CLDC-571

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,14 +1,19 @@
 Rails.application.routes.draw do
   devise_for :admin_users, ActiveAdmin::Devise.config
-  devise_for :users, controllers: { passwords: "users/passwords" }
+  devise_for :users, controllers: { passwords: "users/passwords" }, :skip => [:registrations] 
   devise_scope :user do
     get "confirmations/reset", to: "users/passwords#reset_confirmation"
-  end
+    get 'users/edit' => 'devise/registrations#edit', :as => 'edit_user_registration'
```
Comment (baarkerlounger) [inline config/routes.rb]: Do you actually need this one? Is it not the default?

File: app/views/users/account/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<div class="govuk-grid-row">
```
Comment (baarkerlounger) [inline app/views/users/account/index.html.erb:1]: Can we prettify this one? Indentation looks quite different from the others.

File: app/controllers/users/account_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  def index
+    check_logged_in
+  end
+
+  def personal_details
+    check_logged_in
+  end
+
+  def update
+    if current_user.update('name': params[:user][:name], 'email': params[:user][:email])
```
Comment (baarkerlounger) [inline app/controllers/users/account_controller.rb]: The `permit_params` pattern might be better here? Something like:

```
def update
  if current_user.update(user_params)
    redirect_to(users_account_path)
  else
  # What do we do here?
  end
end

def user_params
  params.require(:user).permit(:email, :name, :password)
end
```

File: app/helpers/account_helper.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module AccountHelper
+  def resource_name
+    :user
+  end
+
+  def resource
+    @resource = current_user
+  end
+
+  def devise_mapping
```
Comment (baarkerlounger) [inline app/helpers/account_helper.rb]: Is this being used anywhere?

File: config/routes.rb (hunk 1 above)
Comment (baarkerlounger) [reply config/routes.rb]: Yes you do...

File: app/helpers/account_helper.rb (hunk 4 above)
Comment (MatthewPhelanMadeTech) [reply app/helpers/account_helper.rb]: The views wouldn't get the resource with out it

File: app/controllers/users/account_controller.rb (hunk 3 above)
Comment (MatthewPhelanMadeTech) [reply app/controllers/users/account_controller.rb]: I'll have a look at it now

**Pull request merged**